event.go

```diff
@@ -70,23 +80,38 @@ func newResponse(res *http.Response, err error, options *Options) *response {
 		}
 	}
 	var body any
+
+	// Throwing an error on network failures causes duplicateBody to segfault on nil
+	if res.Body == nil {
+		res.Body = http.NoBody
+	}
+
 	if options.RecordResponseBody {
 		body, res.Body = duplicateBody(res.Body)
+		body = redactValues(body, options.IncludeSpecifiedResponseBodyKeys)
 	}
 	return &response{
-		Headers:     formatHeaders(res.Header, options.RedactHeaders),
+		Headers:     headersToMap(res.Header),
```

Comment:
Made the decision right now to save Response headers by default (non-redacted), for debugging and billing visibility, but could be convinced to change this back if use case requires it to be redacted.